src/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -238,20 +245,29 @@ public synchronized static NewGameChooserModel getNewGameChooserModel() {
     return s_cachedGameModel;
   }
 
-  // any methods touching s_cachedGameModel should be both static and synchronized
+  /** Starts updating the NewGameChooserModel cache, this can be a lengthy process (seconds) */
   public synchronized static void refreshNewGameChooserModel() {
-    clearNewGameChooserModel();
-    s_cachedGameModel = new NewGameChooserModel();
-  }
+    mapParsingInProgress = true;
+    cacheClearedMessenger = new ClearGameChooserCacheMessenger();
 
-  // any methods touching s_cachedGameModel should be both static and synchronized
-  public synchronized static void clearNewGameChooserModel() {
-    if (s_cachedGameModel != null) {
-      s_cachedGameModel.clear();
-      s_cachedGameModel = null;
+    NewGameChooserModel model = new NewGameChooserModel(cacheClearedMessenger);
+      // Cache might not be populated if another thread has cleared
+      // the cache while we were loading.
+    if (mapParsingInProgress) {
```

Comment:
@DanVanAtta I'm not sure that this works. So the scenario I'm thinking of is as follows:

1. Thread A calls refresh and gets into the if statement but hasn't executed 's_cachedGameModel = model' yet
2. Thread B calls clear and sets s_cachedGameModel to null
3. Thread A resumes and then overrides the clear by setting s_cachedGameModel to the new model